homeassistant/components/todoist/calendar.py

```diff
@@ -222,12 +225,35 @@ async def handle_new_task(call: ServiceCall) -> None:
         if project_id is None:
             raise HomeAssistantError(f"Invalid project name '{project_name}'")
 
+        # Optional section within project
+        section_id: str | None = None
+        if SECTION_NAME in call.data:
+            section_name = call.data[SECTION_NAME]
+            sections = await coordinator.async_get_sections(project_id)
+            for section in sections:
+                if section_name == section.name.lower():
```

Comment:
Let's not add the unnecessary lower-casing to new code.
```suggestion
                if section_name == section.name:
```

Reply:
I've removed `.lower()` from `section_name` as initially asked, because it's already lower-cased by the schema. But because the user-entered data is lower-cased, then the section names from the Todoist API need to be lower-cased for comparison?

Either this `.lower()` needs to remain, or lower-casing should be removed from the schema definition. I am not a fan of the case-insensitive comparison, but since it's happening for the project name I followed the principle of least surprise.

Reply:
You're right, sorry 👍